core/base/src/test/java/org/eclipse/dataspaceconnector/core/security/DefaultPrivateKeyParserTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright (c) 2022 Amadeus
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Amadeus - Initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.core.security;
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class DefaultPrivateKeyParserTest {
+
+    private DefaultPrivateKeyParser parser;
+
+    @BeforeEach
+    public void setUp() {
+        parser = new DefaultPrivateKeyParser();
+    }
+
+    @ParameterizedTest(name = "{index} {1}")
+    @CsvSource({"rsa-privatekey.pem, RSA", "ec-privatekey.pem, EC"})
+    void resolveKey(String keyFileName, String expectedAlgo) {
+        var pem = loadResourceFile(keyFileName);
+
+        var key = parser.parse(pem);
+
+        assertThat(key).isNotNull();
+        assertThat(key.getAlgorithm()).isEqualTo(expectedAlgo);
+    }
+
+    /**
+     * Load content from a resource file.
+     */
+    private static String loadResourceFile(String file) {
+        try (InputStream in = DefaultPrivateKeyParserTest.class.getClassLoader().getResourceAsStream(file)) {
+            Objects.requireNonNull(in);
+            return new BufferedReader(new InputStreamReader(in)).lines().collect(Collectors.joining("\n"));
+        } catch (IOException e) {
+            throw new EdcException(e);
```

Comment:
- no need to rethrow exceptions in tests (just declare them)
- use`java.io.InputStream#readAllBytes` to simplify resource loader code

so you can simplify this code to something like

```
   void resolveKey(String keyFileName, String expectedAlgo) throws IOException {
   ...
    }

    private String loadResourceFile(String file) throws IOException {
        return new String(getClass().getClassLoader().getResourceAsStream(file).readAllBytes());
    }
```